stan/math/prim/prob/gamma_cdf.hpp

```diff
@@ -75,16 +73,19 @@ return_type_t<T_y, T_shape, T_inv_scale> gamma_cdf(const T_y& y,
   using std::exp;
   using std::pow;
 
+  VectorBuilder<!is_constant_all<T_y, T_shape, T_inv_scale>::value,
+                T_partials_return, T_shape>
+      tgamma_vec(size_alpha);
   VectorBuilder<!is_constant_all<T_shape>::value, T_partials_return, T_shape>
-      gamma_vec(size(alpha));
-  VectorBuilder<!is_constant_all<T_shape>::value, T_partials_return, T_shape>
-      digamma_vec(size(alpha));
+      digamma_vec(size_alpha);
 
-  if (!is_constant_all<T_shape>::value) {
-    for (size_t i = 0; i < stan::math::size(alpha); i++) {
+  if (!is_constant_all<T_y, T_shape, T_inv_scale>::value) {
```

Comment:
This is different template logic.

Reply:
This is another situation in which a `tgamma` function could be precomputed for two extra cases (see the `if (!is_constant_all<T_y>::value)` and the `if (!is_constant_all<T_inv_scale>::value)` tests in which before we were recomputing `tgamma(alpha_dbl)`).